impeller/renderer/backend/gles/reactor_gles.cc

```diff
@@ -241,16 +241,23 @@ bool ReactorGLES::FlushOps() {
   SetupDebugGroups();
 #endif
 
-  // Do NOT hold the ops or handles locks while performing operations in case
-  // the ops enqueue more ops.
-  decltype(ops_) ops;
   {
-    Lock ops_lock(ops_mutex_);
-    std::swap(ops_, ops);
-  }
-  for (const auto& op : ops) {
-    TRACE_EVENT0("impeller", "ReactorGLES::Operation");
-    op(*this);
+    // Both the raster thread and the IO thread can flush queued operations.
+    // Ensure that execution of the ops is serialized.
+    Lock execution_lock(ops_execution_mutex_);
```

Comment:
I think a better place for this guard would be around ReactOnce. You can place `IPLR_REQUIRES` around `ReactOnce` to give you static analysis too so its harder to mess up in the future.

Reply:
done